ui/apps/platform/cypress/integration/vulnerabilities/exceptionManagement/approvedDeferralsTable.test.ts

```diff
@@ -0,0 +1,232 @@
+import withAuth from '../../../helpers/basicAuth';
+import { hasFeatureFlag } from '../../../helpers/features';
+import {
+    cancelAllCveExceptions,
+    typeAndSelectCustomSearchFilterValue,
+} from '../workloadCves/WorkloadCves.helpers';
+import {
+    deferAndVisitRequestDetails,
+    visitApprovedDeferralsTab,
+} from './ExceptionManagement.helpers';
+import { approveRequest } from './approveRequestFlow.test';
+import { selectors } from './ExceptionManagement.selectors';
+import { selectors as vulnSelectors } from '../vulnerabilities.selectors';
+
+const comment = 'Defer me';
+const expiry = 'When all CVEs are fixable';
+const scope = 'All images';
+
+describe('Exception Management - Approved Deferrals Table', () => {
+    withAuth();
+
+    before(function () {
+        if (
+            !hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') ||
+            !hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL')
+        ) {
+            this.skip();
+        }
+    });
+
+    beforeEach(() => {
+        if (
+            hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') &&
+            hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL')
+        ) {
+            cancelAllCveExceptions();
+        }
+    });
+
+    after(() => {
+        if (
+            hasFeatureFlag('ROX_VULN_MGMT_WORKLOAD_CVES') &&
+            hasFeatureFlag('ROX_VULN_MGMT_UNIFIED_CVE_DEFERRAL')
+        ) {
+            cancelAllCveExceptions();
+        }
+    });
+
+    it('should be able to view approved deferrals', () => {
+        deferAndVisitRequestDetails({
+            comment,
+            expiry,
+            scope,
+        });
+        approveRequest();
+        visitApprovedDeferralsTab();
+
+        // the deferred request should be approved
+        cy.get(
+            'table td[data-label="Requested action"]:contains("Deferred (when all fixed)")'
+        ).should('exist');
+    });
+
+    it('should be able to navigate to the Request Details page by clicking on the request name', () => {
+        deferAndVisitRequestDetails({
+            comment,
+            expiry,
+            scope,
+        });
+        approveRequest();
+        visitApprovedDeferralsTab();
+
+        const requestNameSelector = 'table td[data-label="Request name"]';
+
+        cy.get(requestNameSelector)
+            .invoke('text')
+            .then((requestName) => {
+                cy.get(requestNameSelector).click();
+                cy.get(`h1:contains("${requestName}")`).should('exist');
+            });
+    });
+
+    it('should be able to sort on the "Request Name" column', () => {
```

Comment:
IMO these sorting tests could benefit from some re-usability in a follow up, and I don't see the harm in having a single test for sorting of this table. Reloading the entire page in order to test each column seems like quite a bit of overhead for tests that should be almost instantaneous.

Reply:
Yeah, they definitely can. I commented above about doing it in an un-optimized manner first. Since there are only 4 tests where we re-use this test, I wanted to create those first, and then deal with refactoring/improvements. I was in the process of making a function for the sort test but thought I should approach it like this first.